### PR TITLE
Implement scan

### PR DESCRIPTION
**Pull request by HarshitaKalani**

### CODE REVIEW

File: integration_tests/intrinsics_189.f90 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    print*, scan(fortran, ao, kind = 4)       
+    if (scan(fortran, ao, kind = 4) /= 2) error stop
+    print*, scan(fortran, ao, .TRUE., 8)  
+    if (scan(fortran, ao, .true., 8) /= 6_8) error stop
+    print*, scan(fortran, c_plus_plus) 
+    if (scan(fortran, c_plus_plus) /= 0) error stop
+
+end
+  
+
```
Comment (Shaikh-Ubaid) [inline integration_tests/intrinsics_189.f90:23]:
```suggestion

end program  
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -5215,10 +5216,6 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 tmp = create_Iachar(x);
             } else if( var_name == "char" ) {
                 tmp = create_StringChr(x);
-            } else if( var_name == "scan" ) {
-                tmp = create_ScanVerify_util(x, var_name);
-            } else if( var_name == "verify" ) {
-                tmp = create_ScanVerify_util(x, var_name);
```
Comment (Shaikh-Ubaid) [inline src/lfortran/semantics/ast_common_visitor.h:5221]: You can also remove the implementation for `create_ScanVerify_util()` if it is unused now.

**Review by Shaikh-Ubaid — COMMENTED** [review]
Shared two comments. Rest looks good to me. Thanks for this. Great work!

**Comment by Shaikh-Ubaid** [comment]
Please mark as "Ready for review" when ready.

File: src/lfortran/semantics/ast_common_visitor.h (hunk 2 above)
Comment (HarshitaKalani) [reply src/lfortran/semantics/ast_common_visitor.h:5221]: Yes, I've already made a separate PR to remove all the implementations which are not required now. Will push that after this gets merged.

**Comment by Pranavchiku** [comment]
Same error, seems unrelated:

```console
+ cd ..
+ git clean -dfx
Removing lf-fast/
Removing lf/
Removing model.dat
+ git checkout -t origin/namelist
fatal: 'origin/namelist' is not a commit and a branch 'namelist' cannot be created from it
```

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 3]
```diff
@@ -98,6 +98,12 @@
             "ret_type_arg_idx": 3
         }
     ],
+    "Scan": [
```
Comment (certik) [inline src/libasr/intrinsic_func_registry_util_gen.py:101]: Let's use some more descriptive name:
```suggestion
    "StringFindSet": [
```

**Review by certik — APPROVED** [review]

**Pull request merged**